src/app/static/src/app/store/modelRun/actions.ts

```diff
@@ -45,13 +46,14 @@ export const actions: ActionTree<ModelRunState, RootState> & ModelRunActions = {
         const {commit, state} = context;
         if (state.status.done) {
             const response = await api<ModelRunMutation, ModelRunMutation>(context)
-                .withSuccess(ModelRunMutation.RunResultFetched)
+                .ignoreSuccess()
                 .withError(ModelRunMutation.RunResultError)
-                .freezeResponse()
                 .get<ModelResultResponse>(`/model/result/${state.modelRunId}`);
 
             if (response) {
                 commit({type: ModelRunMutation.WarningsFetched, payload: response.data.warnings});
+                const frozen = freezer.deepFreeze(response.data);
+                commit({type: ModelRunMutation.RunResultFetched, payload: frozen});
```

Comment:
> I've modified this action because the code in the Stepper to auto-progress was being triggered as soon as the main response was committed (making the step complete), but before the warnings were committed (so the logic to prevent auto-progress in the case of relevant warnings was failing). So we need to commit warnings first (we cannot atomically set warnings in the RunResultFetched mutation, as this mutation is also called when calibrate completes, but it commits warnings to the calibrate module, not the modelRun module).

We've had to work around this a couple of times now. Is it possible that there should actually be a separate mutation for when calibrate completes? I don't know if that would simplify or complicate things!

Reply:
Should be addressed by https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2607. Though the detail in that ticket is a bit sparse, it means we would no longer need to update modelRun state when modeCalibrate completes, and could remove these workarounds. I'll add a little more about that to the ticket before I forget!